test/facade_converter_test.ts

```diff
@@ -113,6 +118,11 @@ describe('type based translation', () => {
           .to.equal(' List < num > x = [ ] ; String y = x . join ( "-" ) ;');
       expectWithTypes('var x: Array<number> = []; var y: string = x.join();')
           .to.equal(' List < num > x = [ ] ; String y = x . join ( "," ) ;');
+      expectWithTypes('var x: Array<number> = []; var y: number = x.find((e) => e == 0);')
+          .to.equal(
+              ' List < num > x = [ ] ; num y = x . firstWhere ( ( e ) => e == 0 , orElse : ( ) => null ) ;');
```

Comment:
these tests are kind of lame. It would be easy to assert we get a certain dart output that doesn't actually work. Perhaps we could do some testing in the e2e tests as well, though on the other hand we don't want to exhaustively cover the cases in that style test.

So clearly I don't have very good guidance here, this is certainly fine for the present PR.